src/data/list/forall2.lean

```diff
@@ -116,6 +116,26 @@ theorem forall₂_length_eq {R : α → β → Prop} :
 | _ _ forall₂.nil          := rfl
 | _ _ (forall₂.cons h₁ h₂) := congr_arg succ (forall₂_length_eq h₂)
 
+theorem forall₂.nth_le :
+  ∀ {x : list α} {y : list β} (h : forall₂ r x y)
+    {i : ℕ} (hx : i < x.length) (hy : i < y.length),
+      r (x.nth_le i hx) (y.nth_le i hy)
+| (a₁ :: l₁) (a₂ :: l₂) (forall₂.cons ha hl) 0        hx hy := ha
+| (a₁ :: l₁) (a₂ :: l₂) (forall₂.cons ha hl) (succ i) hx hy := hl.nth_le _ _
+
+lemma forall₂_of_length_eq_of_nth_le : ∀ {x : list α} {y : list β},
+  x.length = y.length → (∀ i h₁ h₂, r (x.nth_le i h₁) (y.nth_le i h₂)) → list.forall₂ r x y
+| []       []       hl h := forall₂.nil
+| (d₁::l₁) (d₂::l₂) hl h := forall₂.cons
```

Comment:
nit: spaces around `::` seems to be the style used in most of this file; and it's certainly the preferred style since that's how it appears in the goal view.